src/coreComponents/dataRepository/Group.cpp

```diff
@@ -631,16 +638,32 @@ Group const & Group::getBaseGroupByPath( string const & path ) const
     {
       currentGroup = &this->getParent();
     }
+    else if( currentGroup == &NullGroup )
+    {
+      break;
+    }
     else
     {
-      currentGroup = &currentGroup->getGroup( curGroupName );
+      if( hardQuery || currentGroup->hasGroup( curGroupName ) )
+      {
+        currentGroup = &currentGroup->getGroup( curGroupName );
+      }
+      else
+      {
+        currentGroup = &NullGroup;
+      }
     }
   }
   while( currentPosition != string::npos );
 
   return *currentGroup;
 }
 
+bool Group::hasGroupByPath( string const & path ) const
+{
+  return &getBaseGroupByPath( path, false ) != &NullGroup;
```

Comment:
`try except`ing the original `getBaseGroupByPath` and just catching/continuing if it threw also occurred to me, but we throw `std::abort` down the `GEOSX_ERROR` macro hierarchy, which per the spec isn't recoverable.